### PR TITLE
Creates language server protocol for Oberon0

### PR DESCRIPTION
**Pull request by alexandra-hanson**
Creates a language server protocol implementation for Oberon0 using lsp4j. Adds semantic highlighting and go-to declaration features.

I updated the lsp readme at vscode/oberon0lsp/README.md, but I'm not sure if this is sufficient documentation. I also added comments as I thought they were needed but let me know if I should add more / if some comments are missing. It may be that I should add documentation describing the implementation more, but this may be content that goes in the paper write-up associated with my master's project.

### CODE REVIEW

File: grammars/edu.umn.cs.melt.Oberon0/langserver/ReferenceLocations.sv
Comment (alexandra-hanson) [inline grammars/edu.umn.cs.melt.Oberon0/langserver/ReferenceLocations.sv:1]: TO DO: Add more comments to this file?

File: language-server/langserver/src/main/java/edu/umn/cs/melt/oberon0/langserver/Oberon0LanguageServer.java
Comment (alexandra-hanson) [inline language-server/langserver/src/main/java/edu/umn/cs/melt/oberon0/langserver/Oberon0LanguageServer.java]: We probably don't want this initialization grammar to be hard-coded.

File: language-server/langserver/src/main/java/edu/umn/cs/melt/oberon0/langserver/Oberon0LanguageServer.java
Comment (alexandra-hanson) [inline language-server/langserver/src/main/java/edu/umn/cs/melt/oberon0/langserver/Oberon0LanguageServer.java]: Indentation is off here and on lines below.

File: language-server/langserver/src/main/java/edu/umn/cs/melt/oberon0/langserver/Oberon0LanguageService.java
Comment (alexandra-hanson) [inline language-server/langserver/src/main/java/edu/umn/cs/melt/oberon0/langserver/Oberon0LanguageService.java:147]: TODO: There should be more comments in this function.

File: grammars/edu.umn.cs.melt.Oberon0/langserver/ReferenceLocations.sv
Comment (alexandra-hanson) [inline grammars/edu.umn.cs.melt.Oberon0/langserver/ReferenceLocations.sv:61]: Not sure if this is okay. I use this as a workaround to the issue with recursive method declarations, as described in the comment above.

File: grammars/edu.umn.cs.melt.Oberon0/langserver/ReferenceLocations.sv
Comment (alexandra-hanson) [reply grammars/edu.umn.cs.melt.Oberon0/langserver/ReferenceLocations.sv:1]: Done in [11c282b](https://github.com/melt-umn/Oberon0/pull/14/commits/11c282bec77e1a9e14d66328c32e1c324aa91b55)

File: language-server/langserver/src/main/java/edu/umn/cs/melt/oberon0/langserver/Oberon0LanguageService.java
Comment (alexandra-hanson) [reply language-server/langserver/src/main/java/edu/umn/cs/melt/oberon0/langserver/Oberon0LanguageService.java:147]: Done in [11c282b](https://github.com/melt-umn/Oberon0/pull/14/commits/11c282bec77e1a9e14d66328c32e1c324aa91b55)

File: language-server/langserver/src/main/java/edu/umn/cs/melt/oberon0/langserver/Oberon0LanguageServer.java
Comment (krame505) [inline language-server/langserver/src/main/java/edu/umn/cs/melt/oberon0/langserver/Oberon0LanguageServer.java:38]: Style note, it looks like you are using a mix of spaces and tabs for indentation.  We don't really have a prefered style for which one to use, but mixing them makes this file kinda hard to read.

**Review by krame505 — COMMENTED** [review]
Looks generally good to me.  I'll let Eric approve once you've finished writing some more detailed comments/documentation for getting started as we discussed

File: vscode/oberon0lsp/package.json
Comment (alexandra-hanson) [inline vscode/oberon0lsp/package.json]: > 
```suggestion
            "description": "Path to the jar containing an alternate version of the oberon0 driver. This can be a relative path from the first workspace folder."
```

File: language-server/build.sh
Comment (alexandra-hanson) [inline language-server/build.sh:5]: Not sure if I want to be installing just `../oberon0_a5.jar` here

Comment (alexandra-hanson) [reply language-server/build.sh:5]: Actually, I don't believe I need this line at all.

File: language-server/langserver/src/main/java/edu/umn/cs/melt/oberon0/langserver/Oberon0LanguageServer.java
Comment (alexandra-hanson) [reply language-server/langserver/src/main/java/edu/umn/cs/melt/oberon0/langserver/Oberon0LanguageServer.java]: Done in [805e115](https://github.com/melt-umn/Oberon0/pull/14/commits/805e1155bddf92a6c4eabd0f412a8f6a6222e774)

File: language-server/langserver/src/main/java/edu/umn/cs/melt/oberon0/langserver/Oberon0LanguageServer.java
Comment (alexandra-hanson) [reply language-server/langserver/src/main/java/edu/umn/cs/melt/oberon0/langserver/Oberon0LanguageServer.java:38]: Done in [805e115](https://github.com/melt-umn/Oberon0/pull/14/commits/805e1155bddf92a6c4eabd0f412a8f6a6222e774)

**Review by ericvanwyk — APPROVED** [review]

**Pull request merged**
